library/core/src/ptr/mut_ptr.rs

```diff
@@ -404,37 +404,25 @@ impl<T: ?Sized> *mut T {
         if self.is_null() { None } else { Some(unsafe { &*(self as *const MaybeUninit<T>) }) }
     }
 
-    /// Calculates the offset from a pointer.
+    /// Adds an offset to a pointer.
     ///
     /// `count` is in units of T; e.g., a `count` of 3 represents a pointer
     /// offset of `3 * size_of::<T>()` bytes.
     ///
     /// # Safety
     ///
-    /// If any of the following conditions are violated, the result is Undefined
-    /// Behavior:
+    /// The resulting pointer's address is computed as the address of the original pointer plus the
+    /// computed offset, `count * size_of::<T>()`, with infinite precision.
     ///
-    /// * If the computed offset, **in bytes**, is non-zero, then both the starting and resulting
-    ///   pointer must be either in bounds or at the end of the same [allocated object].
-    ///   (If it is zero, then the function is always well-defined.)
+    /// If the following condition is violated, the result is Undefined Behavior:
     ///
-    /// * The computed offset, **in bytes**, cannot overflow an `isize`.
+    /// * If the computed offset is non-zero, then the entire memory range between `self` and the
+    ///   result must be in bounds of the [allocated object] `self` is derived from.
```

Comment:
This sentence sounds like it refers to the allocated object of the pointer itself (`self`), rather than what it is pointing to.

Reply:
In `offset_from` we talk about `self` being derived from a pointer to a particular allocated object... but adding that here makes the sentence too nested, I think. I should probably give a name to the allocated object.